### ISSUE TITLE
Needed GPU Memory

### ISSUE THREAD

**Issue by marcelschilling** [TRI-ML/packnet-sfm#13]
Is there any possibility to use the network with an 4GB GPU device? Thanks in advance.

**Comment by VitorGuizilini-TRI** [comment]
Have you tried with a batch size of 1? If that doesn't work then unfortunately not at this point. 
We have slimmer models for PackNet that should fit, we will soon add those to the repository. We are also working on implementing mixed-precision training, it should address this issue as well.

**Comment by marcelschilling** [comment]
Thanks for your reply. Yes, I tried inference with batch size 1, but unfortunately it doesn‘t work with a 4GB GPU.

**Comment by kwea123** [comment]
Some bugs maybe? Currently the model is initiated without a call of `.eval()`, which means it is in training mode, doesn't that affect batch normalization? Another thing is that in `infer.py`, the decorator `@torch.no_grad()` is missing for `process`. Simply add `with torch.no_grad():` before these lines
https://github.com/TRI-ML/packnet-sfm/blob/f41342d716a72e195a0e72cf12c6b563d182bb91/scripts/infer.py#L74-L75
reduces the memory usage (because it doesn't compute gradient anymore).

As a result, it only consumes 3.6GB for an image of size 384x640 (I use `PackNet01_MR_selfsup_D.ckpt`)
